### ISSUE TITLE
gtk3.20: bug

### ISSUE THREAD

**Issue by megax** [numixproject/numix-gtk-theme#316]
Other:
- Button space:
  ![image](https://cloud.githubusercontent.com/assets/461493/14412221/ee735964-ff5d-11e5-8be0-3aadb4a646c5.png)
  ![image](https://cloud.githubusercontent.com/assets/461493/14412275/02697236-ff5f-11e5-8331-a18ded95de10.png)
- Gedit open file list (border):
  ![image](https://cloud.githubusercontent.com/assets/461493/14451474/aec3b2b6-0087-11e6-98e8-f1a484d86e54.png)
- csd window not border and not separator titlebar/headerbar: Border fixed: https://github.com/numixproject/numix-gtk-theme/pull/379 Separator fixed: https://github.com/numixproject/numix-gtk-theme/pull/381
  Numix:
  ![image](https://cloud.githubusercontent.com/assets/461493/14747015/98eb810c-08b2-11e6-91c9-72e9c68efaef.png)

Adwaita:
![image](https://cloud.githubusercontent.com/assets/461493/14747054/baa3f16c-08b2-11e6-94be-e7790ce4f18e.png)
- Scrollbar style. Fixed: https://github.com/numixproject/numix-gtk-theme/pull/361
- Tooltip style.

Done:
- Separator bug
- Cinnamon-settings
- Font color bug (font changed light color)
- Selected font color and background
- Nemo icon selected backdrop. (black icon)
- Not overshoot.
- file list
- not spinner
- Combobox not pan-down icon.
- Not window decoration. (_window.scss)
- radio, check button (hover...) bug
- Notebook tab bug
- Notebook tab close button size
- double separator????
- baobab progressbar
- titlebar and headerbar border radius bug
- Popover csd border

**Comment by whour** [comment]
![_2016-04-11_10-03-06](https://cloud.githubusercontent.com/assets/4928187/14415059/9d7403fe-ffcd-11e5-87d4-a2b18eb96d85.png)
Firefox are so ugly！
Transparency is too high？

**Comment by eveouyang** [comment]
![_2016-04-11_10-55-51](https://cloud.githubusercontent.com/assets/8010487/14415660/1c585b06-ffd4-11e5-8479-b0c73e59d619.png)
![_2016-04-11_10-27-22](https://cloud.githubusercontent.com/assets/8010487/14415385/24475bd0-ffd1-11e5-9b40-43fb67291940.png)

Numix themes had some problem in Xfce4 with network-manager-applet.
The backgroud color is white and can't see the high line color what I select

The input textbox area and the button theme can't see any more

**Comment by whour** [comment]
@eveouyang Me too. 
So I change to [connman](https://www.archlinux.org/packages/?name=connman) and [CMST](https://aur.archlinux.org/packages/cmst/),network-manager-applet  are too many problems.
Finally，I give up.

**Comment by whour** [comment]
I upgraded  to version `2.5.1.r154.9c9bf64-2` , Firefox bar are normal.
Thanks.

**Comment by megax** [comment]
Fixed bug:
- Nemo icon selected backdrop. (black icon) Fixed: #318
- Font color bug (font changed light color): Fixed: #319
- Not window decoration. (_window.scss) Fixed: #321
- Not overshoot. Fixed: #322 
- Separator. Fixed: https://github.com/shimmerproject/Numix/pull/323

**Comment by marmis85** [comment]
I'm running GNOME 3.20 on Arch and just updated Numix from master, window shadows for GTK3 apps with CSDs are completely gone, and it makes it impossible to resize them with the mouse.
GTK3 apps without CSDs and non-GTK apps seems to be fine (altough they have a smaller shadow than what GTK3 apps used to have)
I've tried both X and Wayland session with the same result.

![screenshot from 2016-04-12 17-12-02](https://cloud.githubusercontent.com/assets/1887387/14465063/cc45de64-00d1-11e6-9f8b-a1d9fefdaee5.png)

**Comment by megax** [comment]
@marmis85 Pull request: https://github.com/shimmerproject/Numix/pull/321

**Comment by marmis85** [comment]
@megax thanks I tried your patch and it works fine

**Comment by marmis85** [comment]
Is it fine if I report GTK 3.20 bugs in this issue or should I open new ones?

The text selection rectangle is missing in text boxes (GtkEntry I think) like in the nautilus rename popover, location bar (CTRL+L) or the epiphany URL bar.

**Comment by SammysHP** [comment]
Another issue: Nautilus with tree view, the separator between the two panes (bookmarks and files) is thin, but the mouse area is thick (sorry, bad description). The invisible area of the separator overlaps the arrows of the tree view.

**Comment by megax** [comment]
@SammysHP Patch: https://github.com/shimmerproject/Numix/pull/330

**Comment by khurshid-alam** [comment]
@megax Old tab style is also a bug which never gets to finish in gtk-3.18. The real old notebook tabs (designed by Satya) had bottom-border (much like many other gtk themes) which makes focus tab identification simpler. The new design in 3.20 has the same issue. (see issue #300 ). For comparison take a look at this screenshot (ignore the top margin bug in the old tab):

![notebook-tab](http://i.imgur.com/RaF83Xm.png)

We also use bottom-border in Numix Light. We would like to keep that consistency.

**Comment by megax** [comment]
@khurshid-alam So you say, we should plan for the older version? The Adwaita theme contains similar to the original conception. I have looked at the program code and I started to make a numix version out of it. I will try to reproduce the original. Should I change the style of the headerbar? If I got the time, I can do something about it.

@lots0logs and @khurshid-alam I try to react to anything you write in time, but please notice, that I can only understand english I could not speak it. My friend is helping me, so I can reply. I try to express myself plain and briefly :)

**Comment by khurshid-alam** [comment]
@megax Yes, we should plan for older version. And no, the headerbar change is **NOT** required as the screenshot of old-tab was taken with Numix-Light.

**Comment by megax** [comment]
@khurshid-alam Okay, thanks.

**Comment by SammysHP** [comment]
@megax Just a big **Thank you!** for your work! It's amazing that you react so fast and with this quality.

**Comment by megax** [comment]
notebook tab patch: https://github.com/shimmerproject/Numix/pull/339

**Comment by SammysHP** [comment]
File dialog still has double separator (here example from Firefox):

![20160414_101524](https://cloud.githubusercontent.com/assets/922874/14524362/3b71c25a-0238-11e6-828d-8c0741643646.png)

**Comment by megax** [comment]
@lots0logs Double separator... idea?

**Comment by megax** [comment]
@lots0logs Arch linux supported .nemo-window. Desktop bug: 
![image](https://cloud.githubusercontent.com/assets/461493/14532125/3fcc83ee-0260-11e6-9970-cfd64425338d.png)
(Right monitor)
magnified:
![image](https://cloud.githubusercontent.com/assets/461493/14532277/d01decf8-0260-11e6-9556-67c0cfafc79e.png)
1 px size white border.

Code bug: https://github.com/shimmerproject/Numix/blob/master/gtk-3.20/scss/apps/_nemo.scss#L112

**Comment by megax** [comment]
@lots0logs https://github.com/shimmerproject/Numix/issues/316#issuecomment-209859348 fixed(??) bug:

```
        statusbar {
            border: 0;
        }
```

Bug source? https://github.com/shimmerproject/Numix/blob/master/gtk-3.20/scss/apps/_gedit.scss#L82

**Comment by TimePower** [comment]
In xfce4(archlinux) running gtk 3.20 apps, get "Theme parsing error".
for example
$gtk3-demo
(gtk3-demo:6281): Gtk-WARNING **: Theme parsing error: gtk.css:158:192: '_text_shadow_color' is not a valid color name

(gtk3-demo:6281): Gtk-WARNING **: Theme parsing error: gtk.css:1609:39: The :inconsistent pseudo-class is deprecated. Use :indeterminate instead.

(gtk3-demo:6281): Gtk-WARNING **: Theme parsing error: gtk.css:1609:77: The :inconsistent pseudo-class is deprecated. Use :indeterminate instead.

(gtk3-demo:6281): Gtk-WARNING **: Theme parsing error: gtk.css:3232:39: 'linear-gradient' is not a valid color name

(gtk3-demo:6281): Gtk-WARNING **: Theme parsing error: gtk.css:3237:41: The :insensitive pseudo-class is deprecated. Use :disabled instead.

(gtk3-demo:6281): Gtk-WARNING **: Theme parsing error: gtk.css:3237:53: The :insensitive pseudo-class is deprecated. Use :disabled instead.

(gtk3-demo:6281): Gtk-WARNING **: Theme parsing error: gtk.css:3244:35: 'border_disabled' is not a valid color name

or

$emacs -q

(emacs:6243): Gtk-WARNING **: Theme parsing error: gtk.css:158:192: '_text_shadow_color' is not a valid color name

(emacs:6243): Gtk-WARNING **: Theme parsing error: gtk.css:1609:39: The :inconsistent pseudo-class is deprecated. Use :indeterminate instead.

(emacs:6243): Gtk-WARNING **: Theme parsing error: gtk.css:1609:77: The :inconsistent pseudo-class is deprecated. Use :indeterminate instead.

(emacs:6243): Gtk-WARNING **: Theme parsing error: gtk.css:3232:39: 'linear-gradient' is not a valid color name

(emacs:6243): Gtk-WARNING **: Theme parsing error: gtk.css:3237:41: The :insensitive pseudo-class is deprecated. Use :disabled instead.

(emacs:6243): Gtk-WARNING **: Theme parsing error: gtk.css:3237:53: The :insensitive pseudo-class is deprecated. Use :disabled instead.

(emacs:6243): Gtk-WARNING **: Theme parsing error: gtk.css:3244:35: 'border_disabled' is not a valid color name

(emacs:6243): Gtk-WARNING **: Theme parsing error: gtk-keys.css:108:18: The 'gtk-key-bindings' property has been renamed to '-gtk-key-bindings'

(emacs:6243): Gtk-WARNING **: Theme parsing error: gtk-keys.css:112:18: The 'gtk-key-bindings' property has been renamed to '-gtk-key-bindings'

(emacs:6243): Gtk-WARNING **: Theme parsing error: gtk-keys.css:116:18: The 'gtk-key-bindings' property has been renamed to '-gtk-key-bindings'

(emacs:6243): Gtk-WARNING **: Theme parsing error: gtk-keys.css:120:18: The 'gtk-key-bindings' property has been renamed to '-gtk-key-bindings'

**Comment by SammysHP** [comment]
But why is it there? `scss/apps/*` should contain _only_ definitions for specific apps.

**Comment by lots0logs** [comment]
@SammysHP Not all apps have been updated with new 3.20 style classes so we're having to work-around that. As the apps get updated with style classes we will encapsulate their styles again.

**Comment by megax** [comment]
@SammysHP Fixed double separator bug: https://github.com/numixproject/numix-gtk-theme/pull/375

**Comment by megax** [comment]
@lots0logs Nemo button separator bug:

Now:
![image](https://cloud.githubusercontent.com/assets/461493/14743531/3d1e05f8-08a2-11e6-9f44-62f999ea260a.png)

Normal style:
![image](https://cloud.githubusercontent.com/assets/461493/14743567/6bd840fc-08a2-11e6-9451-cd833ecf9a99.png)

Code bug: https://github.com/numixproject/numix-gtk-theme/blob/master/gtk-3.20/scss/apps/_nemo.scss#L131

```
            button {
                @include button($bg_color, $fg_color);
            }
```

**Comment by megax** [comment]
Other gtk3.20 bug:
https://github.com/numixproject/numix-gtk-theme/issues/340
https://github.com/numixproject/numix-gtk-theme/issues/346
https://github.com/numixproject/numix-gtk-theme/issues/351
https://github.com/numixproject/numix-gtk-theme/issues/357
https://github.com/numixproject/numix-gtk-theme/issues/359
https://github.com/numixproject/numix-gtk-theme/issues/362
https://github.com/numixproject/numix-gtk-theme/issues/370

**Comment by megax** [comment]
Window shadow bug: https://github.com/numixproject/numix-gtk-theme/commit/3cc526c0d6b1a59049b3d91ba490fb66b5281f1c

**Comment by SammysHP** [comment]
I think it's time for merging pull requests. ;)

There are two conflicting PR: #360 from @megax and #361 from me. I suggest to merge #361, not because it's mine (:wink:), but because I refactored the whole scrollbar code and made the steppers working (although they are disabled in Numix).

I used my changes for a few days and didn't notice any problems (https://github.com/SammysHP/numix-gtk-theme/tree/tmpmerge). I'll also merge the changes from @megax into my branch to test them.
